docs/cli-usage.md

```diff
@@ -0,0 +1,96 @@
+## Using Kubernetes from the command line
+
+A how to guide for accessing and running applications on a kubernetes cluster.
+
+### Basic Setup
+The kubernetes cluster is accessed via the ```kubectl`` command.  Full details of using the ```kubectl``` command are described [elsewhere](docs/cli.md).  Make sure that ```kubectl``` is in your path by running
+
+```sh
+which kubectl
+```
+
+If that command returns a path to the binary, you are ready to go.  If it doesn't, you need to update your PATH to include the appropriate directory from the Kubernetes directory.
+
+### Commands
+#### run
+```sh
+kubectl run <pod-name> --image=nginx [--replicas=N] [--export [--external]]
+```
+
+This translates into a call to create a replication controller and optionally, a call to create a service which exposes the pods created by that replication controller.
+
+
+#### describe
+```sh
+kubectl describe <name>
+```
+
+Describe attempts a get against the three resource types (Pods, Replication Controllers, Services) and returns matching resource(s) for the named resource.  It ignores 404 errors, unless all of the get requests return 404.
+
+
+#### resize
+```sh
+kubectl resize <name> --replicas=N
+```
+
+Update is used to either update the size of a replication controller.
+
+#### export
+```sh
+kubectl export <service-name> <name> [--external] [--public-ip=<ip>] [--create-balancer=<bool>]
+```
+
+Export is used to create a service which represents a replicated set of pods.  First the replication controller is looked up using <name> then a new service called <service-name> is created.  If --external is specified, the service is also exposed on a public IP address, if --public-ip is specified, it is used as the ip address.  If --create-balancer is true (the default) an attempt to create a cloud load balancer with that IP address is performed.
```

Comment:
That suggests a problem with the term for service, IMO.

However, if what you want is a name for a service generator, akin to `run`, `expose` isn't bad.

Extracting general generator requirements from #1695:

* It needs to be possible to access same generator functionality declaratively as is available imperatively using the CLI. A configuration generator will therefore be at the core of the approach. Configuration generation was first discussed in #1007 and #1694. The command-line arguments for the imperative command should be used to generate the config generator input schema, with as much of the intelligent defaulting inside the config generator (or even in the API e.g., #2643) as possible. 
* The config generator schema needs to be versioned. Otherwise, it will become unchangeable -- we've had this problem internally in spades. The default version can just be the latest one, but anyone who cares about reproducibility should be able to access any supported version of the generator via flag. That probably means the interpretation of the command-line flags needs to be versioned inside kubectl for this subcommand.
* It needs to be straightforward to discover and manage (update, delete) the generated objects, using either the generator schema or the native objects.
* It needs to be possible to just generate the primitive API objects without sending them to the apiserver, to facilitate a transition to declarative management.
* The standard kubectl functionality for the primitive objects should be used to communicate with the API, not custom client calls.
* There should be a path towards accessing a non-trivial amount of the power of the API, though via somewhat opinionated idioms. 
* Other planned config and deployment mechanisms should compose cleanly with it.

Not all of this needs to be supported out the gate, but whatever we build needs to be a compatible subset. For instance, we should decide how to specify schema fields on the command line, what flags we should use for a dry run (generation only), etc.

That said, what value does this add? Does it just generate the service name? It looks to me like a general mechanism for specifying object fields on the command line would provide the vast majority of the value -- avoiding having to create a file. There should be a way to do that for every type of object.